src/status_im/contexts/wallet/common/sheets/account_options/view.cljs

```diff
@@ -73,9 +75,28 @@
                                               {:type :positive
                                                :text (i18n/label :t/address-copied)}])
                                 (clipboard/set-string address))}
+        {:icon                :i/qr-code
+         :accessibility-label :show-address-qr
+         :label               (i18n/label :t/show-address-qr)
+         :on-press            #(rf/dispatch [:open-modal :wallet-share-address {:status :share}])}
         {:icon                :i/share
          :accessibility-label :share-account
-         :label               (i18n/label :t/share-account)}
+         :label               (i18n/label :t/share-account)
+         :on-press            (fn []
```

Comment:
@mmilad75 I think this is too much logic on the view. Ideally, our views should just dispatch something.

We just need to define an *event* and create a `:share/open` *effect* instead of calling `(share/open ...)` directly. That's the re-frame architecture we should try to abide to, immutable events, as simple as we can make views.

```clojure
(ns status-im.contexts.wallet.common.sheets.account-options.events
  (:require
    [react-native.platform :as platform]
    [utils.re-frame :as rf]))

(rf/reg-event-fx :wallet/share-account
 (fn [_ [{:keys [content title]}]]
   {:fx [[:dispatch [:hide-bottom-sheet]]
         [:share/open ; => This needs to be defined, but it will just wrap the call to react-native-share
          (if platform/ios?
            {:activityItemSources
             [{:placeholderItem {:type    "text"
                                 :content content}
               :item            {:default {:type    "text"
                                           :content content}}
               :linkMetadata    {:title title}}]}
            {:title   title
             :subject title
             :message content})]]}))
```

And in the view:

```clojure
[quo/action-drawer
 [[...
   {:icon                :i/share
    :accessibility-label :share-account
    :label               (i18n/label :t/share-account)
    :on-press            (fn []
                           (js/setTimeout
                            #(rf/dispatch [:wallet/share-account
                                           {:title share-title :content address}])
                            600))}
   ...]]]
```

Reply:
Done